### PR TITLE
fix: Propagate filter params for locator filters using containsElement

### PR DESCRIPTION
**Pull request by jimevanssfdc**

### CODE REVIEW

File: utam-compiler/src/main/java/utam/compiler/representation/ElementMethod.java [hunk 1]
```diff
@@ -296,6 +300,15 @@ public Filtered(
       this.unionType = asUnionTypeOrNull(implType);
     }
 
+    private void setFilterParameterClassImports() {
+      // For non-primitive method parameters added by a filter, we must add class imports
+      // for those types or the generated code will not be compilable by javac.
+      parametersTracker.getMethodParameters().stream()
+          .filter(param -> Arrays.stream(PrimitiveType.values())
+              .noneMatch(primitiveType -> primitiveType.isSameType(param.getType())))
+          .forEach(param -> ParameterUtils.setImport(classImports, param.getType()));
```
Comment (lizaiv77) [inline utam-compiler/src/main/java/utam/compiler/representation/ElementMethod.java:309]: filter is redundant, ParameterUtils.setImport is not going to add imports for primitives

Comment (jimevanssfdc) [reply utam-compiler/src/main/java/utam/compiler/representation/ElementMethod.java:307]: I didn't notice that. Clever. That simplifies things quite a bit. Fixed in next commit.

Comment (lizaiv77) [reply utam-compiler/src/main/java/utam/compiler/representation/ElementMethod.java:307]: sorry, forgot but we should filter for parameter not being literal:
```
parametersTracker.getMethodParameters().
          stream().filter(p -> !p.isLiteral())
          .forEach(p -> setImport(classImports, p.getType()));
```

Comment (jimevanssfdc) [reply utam-compiler/src/main/java/utam/compiler/representation/ElementMethod.java:307]: No, I'm afraid that doesn't work. Consider:
```json
{
  "elements": [
    {
      "name": "test",
      "public": true,
      "selector": {
        "css": "my-custom-element[with-attribute='attr-value']",
        "returnAll": true
      },
      "filter": {
        "apply": "containsElement",
        "args": [
          {
            "type": "locator",
            "value": {
              "css": "input[value='%s']",
              "args": [
                {
                  "name": "value",
                  "type": "string"
                }
              ]
            }
          }
        ],
        "findFirst": true,
        "matcher": {
          "type": "isTrue"
        }
      }
    }
  ]
}
```
The `locator` parameter of the `containsElement` action for the filter _is_ a literal. If we filter on not-literal, we won't add the `LocatorBy` import, which is one of the whole reasons for this PR in the first place.

**Review by lizaiv77 — APPROVED** [review]

**Pull request merged**